lib/base.rb

```diff
@@ -25,14 +25,18 @@ def information(key, value)
     @information[key] = value
   end
 
-  def run!(request, response)
+  def run!(request, response, test_hook)
+    run_all = self.class::Conf.run_all
     if @skip
       @skip = false
       return
     end
-    add_test_case(request, response)
+    if run_all
+      add_test_case(request, response)
+    else
+      add_test_case(request, response) unless test_hook
```

Comment:
Scratch that, I remember why this is set up this way. When you change the run_all config, `SmarfDoc.run!(request, response)` is still in the `spec_helper`. I needed a way to differentiate between the test hook call and when the user wants to explicitly run a test. The solution was to create an optional `test_hook` boolean that defaults to `false`, but is set `true` by the test suite calls.